### PR TITLE
Add activityId to response and custom executeRequest

### PR DESCRIPTION
**Pull request by southpolesteve**

### CODE REVIEW

File: src/CosmosClientOptions.ts [hunk 1]
```diff
@@ -31,4 +38,8 @@ export interface CosmosClientOptions {
    * Use an agent such as https://github.com/TooTallNate/node-proxy-agent if you need to connect to Cosmos via a proxy
    */
   agent?: Agent;
+  /** An optional function to be called on each request
+   * // TODO Add docs
+   */
+  executeRequest?: ExecuteRequest;
```
Comment (southpolesteve) [inline src/CosmosClientOptions.ts]: @christopheranderson I'm definitely not set on this API. I would like to maybe rethink it when I do the pipeline work but for now, I need something here to get this into the portal

Comment (christopheranderson) [reply src/CosmosClientOptions.ts]: Why do we need this executeRequest thing?

File: src/request/FeedResponse.ts [hunk 2]
```diff
@@ -16,4 +16,7 @@ export class FeedResponse<TResource> {
   public get requestCharge(): number {
     return this.headers[Constants.HttpHeaders.RequestCharge];
   }
+  public get activityId(): number {
```
Comment (christopheranderson) [inline src/request/FeedResponse.ts]: Please something to our tests to make sure it is being set in single and cross partition scenarios.

Comment (christopheranderson) [inline src/request/FeedResponse.ts]: string
```suggestion
  public get activityId(): string{
```

File: src/CosmosClientOptions.ts (hunk 1 above)
Comment (christopheranderson) [reply src/CosmosClientOptions.ts]: Is this just needed for the portal? Could we do it in a hacky way where we expose it as a public type on ClientContext so you can override it?

**Pull request merged**